relay-general/src/store/normalize.rs

```diff
@@ -1005,6 +1018,68 @@ impl<'a> Processor for NormalizeProcessor<'a> {
     }
 }
 
+/// If the logger is longer than [`MaxChars::Logger`], it returns a String with
+/// a shortened version of the logger. If not, the same logger is returned as a
+/// String.
+///
+/// To shorten the logger, all extra chars that don't fit into the maximum limit
+/// are removed, from the beginning of the logger.  Then, if the remaining
+/// substring contains a `.` somewhere but in the end, all chars until `.`
+/// (exclusive) are removed.
+///
+/// Additionally, the new logger is prefixed with `*`, to indicate it was
+/// shortened.
+fn shorten_logger(logger: &str) -> String {
+    let logger_len = bytecount::num_chars(logger.as_bytes());
+    if logger_len <= MaxChars::Logger.limit() {
+        return logger.to_owned();
+    }
+
+    let mut tokens = logger.split("").collect_vec();
+    // Remove empty str tokens from the beginning and end.
+    tokens.pop();
+    tokens.reverse(); // Prioritize chars from the end of the string.
+    tokens.pop();
+
+    remove_logger_extra_chars(&mut tokens);
+    remove_logger_word(&mut tokens);
```

Comment:
Shouldn't this be called only if `remove_logger_extra_chars` cut a word in half? If my logger is `foo.bar`, wouldn't `remove_logger_word` crop it to `.bar`?